### ISSUE TITLE
Move Parkingspace and Workspace to Space

### ISSUE THREAD

**Issue by jbkoh** [RealEstateCore/rec#282]
Currently both ParkingSpace and Workspace are subclasses of Zone. However, similar to Rooms, both ParksingSpace and Workspace have some space codes and treated as the same abstraction level of Rooms in floorplans. A counter example is the Office class. Office is an area which can be a room or an open area, which is currently a subclass of Room. I think Office and Workspace should be in the same subtree of the class hierarchy instead of one being a Zone and the other one being a Room.

[IFC's definition of space](https://standards.buildingsmart.org/IFC/RELEASE/IFC4_3/HTML/lexical/IfcSpace.htm) is I think very informative, which Brick originally adopted.
> Spaces are areas or volumes that provide for certain functions within a building.

Based on this definition, Workspace is serving a purpose and thus should be a Space. HVAC_Zone is auxiliary to the purpose of the corresponding space, and thus should be a Zone.

Another drastic solution would be merging the concept of Space and Zone because the distinction is often not clear and not useful. In fact, [IFC defines zones](https://standards.buildingsmart.org/IFC/RELEASE/IFC4_3/HTML/lexical/IfcZone.htm) to be "A zone is a group of spaces, partial spaces or other zones.", which may hint that we need to be able to arbitrarily represent a part or a collection of spaces, and if a zone can be a part of a space, maybe just call them all spaces instead.

In summary, **my preference is to clearly define Space based on IFC and move Workspace/ParkingSpace out of Zone based on that**. Another possible, but less preferred, option is to merge the concept of Zone/Space to make Space more generic
